stubs/docker/docker/_types.pyi

```diff
@@ -1,8 +1,15 @@
 # Internal-use module for types shared by multiple modules.
 # This does not match a module in docker-py.
 
+from pathlib import Path
+from typing import TypedDict, type_check_only
 from typing_extensions import TypeAlias
 
 # Type alias for JSON, explained at:
 # https://github.com/python/typing/issues/182#issuecomment-1320974824.
 JSON: TypeAlias = dict[str, JSON] | list[JSON] | str | int | float | bool | None
+
+@type_check_only
+class ContainerWeightDevice(TypedDict):
```

Comment:
See my comment on the other PR. We should prefix this with an underscore, to also make it clear that this is an internal class in places where this is used.

Reply:
Thank you. I will pause working on this PR until that other discussion is resolved.